wled00/FX.cpp

```diff
@@ -94,6 +95,50 @@ uint16_t mode_static(void) {
 static const char _data_FX_MODE_STATIC[] PROGMEM = "Solid";
 
 
+/*
+ * Copy selected segment and perform (optional) color adjustments
+ */
+// TODO: what is the correct place to put getRenderedPixelXY()?
+static uint32_t getRenderedPixelXY(Segment& seg, unsigned x, unsigned y = 0) {
+  // We read pixels back following mirror/reverse/transpose but ignoring grouping
+  // For every group-length pixels, add spacing
+  x *= seg.groupLength(); // expand to physical pixels
+  y *= seg.groupLength(); // expand to physical pixels
+  if (x >= seg.width() || y >= seg.height()) return 0;  // fill out of range pixels with black
+  #warning this check can be removed once 2D offset is fixed
```

Comment:
FYI `offset` field is *unused* AKA reserved if the segment is 2D.

Reply:
ok, can remove the #warning then and keep that if statement.